pkg/installer/packages/manager.go

```diff
@@ -0,0 +1,300 @@
+	}
+
+	pkg, err := m.downloader.Download(ctx, url)
+	if err != nil {
+		return fmt.Errorf("could not download package: %w", err)
+	}
+	tmpDir, err := os.MkdirTemp(m.tmpDirPath, fmt.Sprintf("install-stable-%s-*", pkg.Name)) // * is replaced by a random string
+	if err != nil {
+		return fmt.Errorf("could not create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	configDir := filepath.Join(m.configsDir, pkg.Name)
+	err = extractPackageLayers(pkg.Image, configDir, tmpDir)
+	if err != nil {
+		return fmt.Errorf("could not extract package layers: %w", err)
+	}
+	err = m.repositories.Create(ctx, pkg.Name, pkg.Version, tmpDir)
+	if err != nil {
+		return fmt.Errorf("could not create repository: %w", err)
+	}
+	return m.setupUnits(ctx, pkg.Name)
+}
+
+// InstallExperiment installs an experiment on top of an existing package.
+func (m *managerImpl) InstallExperiment(ctx context.Context, url string) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+	err := utils.CheckAvailableDiskSpace(mininumDiskSpace, m.packagesPath)
+	if err != nil {
+		return fmt.Errorf("not enough disk space: %w", err)
+	}
+
+	pkg, err := m.downloader.Download(ctx, url)
+	if err != nil {
+		return fmt.Errorf("could not download package: %w", err)
+	}
+	tmpDir, err := os.MkdirTemp(m.tmpDirPath, fmt.Sprintf("install-experiment-%s-*", pkg.Name)) // * is replaced by a random string
+	if err != nil {
+		return fmt.Errorf("could not create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	configDir := filepath.Join(m.configsDir, pkg.Name)
+	err = extractPackageLayers(pkg.Image, configDir, tmpDir)
+	if err != nil {
+		return fmt.Errorf("could not extract package layers: %w", err)
+	}
+	repository := m.repositories.Get(pkg.Name)
+	err = repository.SetExperiment(ctx, pkg.Version, tmpDir)
+	if err != nil {
+		return fmt.Errorf("could not set experiment: %w", err)
+	}
+	return m.startExperiment(ctx, pkg.Name)
+}
+
+// RemoveExperiment removes an experiment.
+func (m *managerImpl) RemoveExperiment(ctx context.Context, pkg string) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	repository := m.repositories.Get(pkg)
+	err := repository.DeleteExperiment(ctx)
+	if err != nil {
+		return fmt.Errorf("could not delete experiment: %w", err)
+	}
+	return m.stopExperiment(ctx, pkg)
+}
+
+// PromoteExperiment promotes an experiment to stable.
+func (m *managerImpl) PromoteExperiment(ctx context.Context, pkg string) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	repository := m.repositories.Get(pkg)
+	err := repository.PromoteExperiment(ctx)
+	if err != nil {
+		return fmt.Errorf("could not promote experiment: %w", err)
+	}
+	return m.stopExperiment(ctx, pkg)
+}
+
+// Remove uninstalls a package.
+func (m *managerImpl) Remove(ctx context.Context, pkg string) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	err := m.repositories.Delete(ctx, pkg)
+	if err != nil {
+		return fmt.Errorf("could not remove package: %w", err)
+	}
+	return nil
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** Shouldn't we call a package-specific removal method here in `Remove`? For the injector removing the ld.preload.so config is essential. It should be done before deleting the package files on disk, fwiw.